test/integration/producer/message_options_test.exs

```diff
@@ -0,0 +1,107 @@
+defmodule Pulsar.Integration.Producer.MessageOptionsTest do
+  use ExUnit.Case, async: true
+
+  alias Pulsar.Test.Support.System
+  alias Pulsar.Test.Support.Utils
+
+  @moduletag :integration
+  @client :producer_message_options_test_client
+  @topic "persistent://public/default/producer-message-options-test"
+
+  setup_all do
+    broker = System.broker()
+
+    :ok = System.create_topic(@topic)
+
+    {:ok, _client_pid} =
+      Pulsar.Client.start_link(
+        name: @client,
+        host: broker.service_url
+      )
+
+    {:ok, group_pid} =
+      Pulsar.start_producer(@topic,
+        client: @client,
+        name: "message-options-producer"
+      )
+
+    [producer] = Pulsar.get_producers(group_pid)
+
+    :ok =
+      Utils.wait_for(fn ->
+        state = :sys.get_state(producer)
+        state.producer_name != nil
+      end)
+
+    on_exit(fn ->
+      Pulsar.stop_producer(group_pid)
+      Pulsar.Client.stop(@client)
+    end)
+
+    %{producer: group_pid}
+  end
+
+  test "send message with partition key", %{producer: producer} do
+    assert {:ok, _message_id} =
+             Pulsar.send(producer, "payload with key", key: "user-123", client: @client)
+  end
+
+  test "send message with ordering key", %{producer: producer} do
+    assert {:ok, _message_id} =
```

Comment:
Same as above. Maybe, we could consume the message and assert that the ordering key is there in the message metadata?